### ISSUE TITLE
Failing to create new database due to unavailable region combination

### ISSUE THREAD

**Issue by ramielrowe** [datastax/terraform-provider-astra#19]
Error: Error: cloud provider and region combination not available: GCP/us-east1

If I manually curl `https://api.astra.datastax.com/v2/availableRegions` with my token, I don't see any `serverless` tiers listed. But, if I go to the web console, I can create databases in GCP/us-east1.

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

### Affected Resource(s)
astra_database

### Terraform Configuration Files
```hcl
provider "astra" {}

terraform {
    required_providers {
        astra = {
            source = "datastax/astra"
            version = "0.0.3-pre"
        }
    }
}

resource "astra_database" "temporal_db" {
    name           = var.name
    keyspace       = "temporal"
    cloud_provider = "GCP"
    region         = "us-east1"
}

data "astra_secure_connect_bundle_url" "temporal_db" {
    database_id = astra_database.temporal_db.id
}
```

### Debug Output
NA

### Panic Output
NA

### Expected Behavior
Database created in us-east1

### Actual Behavior
Error: cloud provider and region combination not available: GCP/us-east1

### Steps to Reproduce
1. Create Database Admin token
2. Set that token to ASTRA_API_TOKEN
3. terraform apply

**Comment by phact** [comment]
There are restrictions but they are accurately reflected in `availableRegions`

```
$ ./terraform show | grep -a2 serverless
            cloud_provider = "AWS"
            region         = "ap-southeast-1"
            tier           = "serverless"
        },
        {
            cloud_provider = "AWS"
            region         = "eu-central-1"
            tier           = "serverless"
        },
        {
            cloud_provider = "AWS"
            region         = "us-east-1"
            tier           = "serverless"
        },
        {
            cloud_provider = "GCP"
            region         = "asia-south1"
            tier           = "serverless"
        },
        {
            cloud_provider = "GCP"
            region         = "europe-west1"
            tier           = "serverless"
        },
        {
            cloud_provider = "GCP"
            region         = "us-east1"
            tier           = "serverless"
        },
        {
            cloud_provider = "AWS"
            region         = "us-west-2"
            tier           = "serverless"
        },
        {
            cloud_provider = "GCP"
            region         = "us-west1"
            tier           = "serverless"
        },
        {
            cloud_provider = "AZURE"
            region         = "westus2"
            tier           = "serverless"
        },
        {
            cloud_provider = "AZURE"
            region         = "westeurope"
            tier           = "serverless"
        },
        {
            cloud_provider = "AZURE"
            region         = "australiaeast"
            tier           = "serverless"
        },
        {
```

If you're not seeing serverless available when you hit the endpoint you may be using a classic only Astra organization. You can go in the Astra UI and create a new org and select `on demand`.

**Comment by ramielrowe** [comment]
Hi,

The organization I'm using was created automatically via the GCP marketplace. I don't think we have any control over it. I can create serverless databases via the web UI. It just doesn't work for terraform/API.

**Comment by phact** [comment]
I see, there are two GCP marketplace offerings. You may be using the old one. 
![image](https://user-images.githubusercontent.com/1313220/118704127-cb874b00-b7e4-11eb-949d-57232b7b9a4f.png)
Please make sure you pick the orange one, not the black one. We've been working with google to remove the old one but it is taking a long time. Apologies for this.

**Comment by ramielrowe** [comment]
![image](https://user-images.githubusercontent.com/1068858/118705682-83692800-b7e6-11eb-94a9-2c3d914cddf4.png)
We used the orange one.

**Comment by vigohe** [comment]
any comments? We also have the same issue with azure, no serverless tiers, we selected the orange one

**Comment by phact** [comment]
Should be fixed in v1.0.4+

**Comment by harishdesetti1206** [comment]
Great @phact the issue is fixed.. i used the version 1.0.6 and it worked like charm .. Thanks .!
